#!/usr/bin/env python
PACKAGE = "flir_gige"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# name / type / level / description / default / min / max
gen.add("thresh", int_t, 0, "pixel threshold for blob detection", 200, 0, 255)
gen.add("sigma", double_t, 0, "gaussian sigma", 0.8, 0.6, 6.0)
gen.add("min_area", double_t, 0, "Smallest area found", 100, 1, 200)
gen.add("max_area", double_t, 0, "Largest area found", 200, 200, 900)
gen.add("max_width_height_distortion", double_t, 0, "Max width height distortion", 0.3, 0, 1)
gen.add("max_circular_distortion", double_t, 0, "Max deviation from roundness", 0.3, 0, 1)
exit(gen.generate(PACKAGE, "flir_gige", "CalibProcDyn"))
